added how to guides on: 1. assigning more than four states to data 2.… (#4087)
influxdata/docs-v2: content/resources/how-to-guides/assigning-more-than-four-states.md

---
title: Assign custom states to data
description: >
  Learn how overcome a limitation of the `monitor.stateChanges()` function and assign custom states to your data. 
menu:
  resources:
    parent: How-to guides
weight: 101
---
## Problem

> Users want to use the monitor package and take advantage of functions like [monitor.stateChangesOnly(](flux/v0.x/stdlib/influxdata/influxdb/monitor/statechangesonly/). However, the monitor.stateChangesOnly() function only allows you to monitor four states: "crit", "warn", "ok", and "info". Users want to be able to assign and monitor state changes across more than four states or define custom states. 

Reviewer: ```suggestion
You may want to use the [`monitor` package](/flux/v0.x/stdlib/influxdata/influxdb/monitor/) and take advantage of functions like [monitor.stateChangesOnly()](flux/v0.x/stdlib/influxdata/influxdb/monitor/statechangesonly/). However, `monitor.stateChangesOnly()` only allows you to monitor four states: "crit", "warn", "ok", and "info". What if you want to be able to assign and monitor state changes across custom states or more than four states?
```
